### ISSUE TITLE
Nodes Deform Support

### ISSUE THREAD

**Issue by fxtressia** [Inochi2D/inox2d#5]
This might be the most complicated part of Inochi2D...

Upstream reference: https://github.com/Inochi2D/inochi2d/tree/main/source/inochi2d/core/nodes

**Comment by fxtressia** [comment]
@linkmauve Hellooo

Do you want to add any comments here? It's quite hard to implement this one, so we would appreciate any help.

Also, I would appreciate a lot if you could push your WIP changes to your implementation repository. It would help us a lot with understanding how to implement things.

**Comment by Speykious** [comment]
> This might be the most complicated part of Inochi2D...

I'd say the most complicated part in retrospect was handling parameters. In fact, when I did it, I didn't read the reference implementation. Or well, I tried, and just couldn't understand what's happening. So instead, when I asked what was the math behind it, I got "bi-interpolation" as an answer, searched what kind of thing it was, and extrapolated everything that had to happen from there. And I gotta say, given how this is coded right now, I'm glad I didn't read the reference implementation, because I probably got to an implementation that was way cleaner for Rust than if I had tried to copy the D code. :D
